hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -165,8 +164,8 @@ object HoodieSparkSqlWriter {
     val sparkContext = sqlContext.sparkContext
 
     val fs = basePath.getFileSystem(sparkContext.hadoopConfiguration)
-    tableExists = fs.exists(new Path(basePath, HoodieTableMetaClient.METAFOLDER_NAME))
-    var tableConfig = getHoodieTableConfig(sparkContext, path, mode, streamingWritesParamsOpt.map( _.hoodieTableConfigOpt).orElse(Option.apply(Option.empty)).get)
+    var tableExists = fs.exists(new Path(basePath, HoodieTableMetaClient.METAFOLDER_NAME))
+    var tableConfig = getHoodieTableConfig(sparkContext, path, mode, streamingWritesParamsOpt.map( _.hoodieTableConfigOpt).orElse(Option.apply(Option.empty)).get, tableExists)
```

Comment:
I'm confused, the `HoodieSparkSqlWriter` is not designed as thread safe basically.

Reply:
> I'm confused, the HoodieSparkSqlWriter is not designed as thread safe basically.

We will create multiple writers concurrently in our application like this: 

```java
DataFrameWriter<Row> writer =
            spark.createDataFrame(javaRDD, dataset.schema()).toDF().write();
writer.format("org.apache.hudi").mode(SaveMode.Overwrite).save(targetPath);
```

So we expect `HoodieSparkSqlWriter` to be thread-safe...

Reply:
Did you ever check the other potential shared `states` in the write client then? Are they all thread safe?

Reply:
In fact, there may still exist some potential concurrent issues when we write multiple tables concurrently using Spark. For example, in our inner hudi version (based on 0.11.x), the key-generator use table name to register spark UDF and when we have tables with the same name between different databases we may generate a  wrong key. The problem is solved in the latest branch. 

Writing multiple hudi tables concurrently in a single spark job is a very common use case, we need to solve any concurrent issue met.